### PR TITLE
Add documentation for embedded/nested documents query

### LINKED ISSUE #1310: Add documentation on embedded documents and dot notation support for arrays
### What should be done?

Add a section on document queries, embedded documents, array queries, and Dot notation support for arrays. 

* [x] add section for dot notations in the document section
* [x] retrieve documents arrays using dot notation
* [x] update documents arrays using dot notation
* [x] query an embedded document using dot notation
* [x] update an embedded document using dot notation

### PR DESCRIPTION
**Pull request by Fashander**
# Description

Closes #1310.

## Readiness checklist

* [ ] I added tests for new functionality or bugfixes.
* [ ] I ran `task all`, and it passed.
* [ ] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: website/docs/understanding_ferretdb.md [hunk 1]
```diff
@@ -82,8 +82,8 @@ To reference the fourth field in the array, use the dot notation `"animals.3"`.
 
 Here are more examples of dot notations on arrays:
 
-* [Query an array](https://docs.ferretdb.io/Basic_operations/read/#retrieve-documents-containing-a-value-in-an-array)
-* [Update an array](https://docs.ferretdb.io/Basic_operations/update/#update-an-array-element)
+* [Query an array](./basic_operations/read/#retrieve-documents-containing-a-specific-value-in-an-array)
```
Comment (chilagrow) [inline website/docs/understanding_ferretdb.md:85]:
```suggestion
* [Query an array](./basic_operations/read.md#retrieve-documents-containing-a-specific-value-in-an-array)
```

Comment (Fashander) [reply website/docs/understanding_ferretdb.md:85]: It doesn't work with a `.md` added to it. This is an example: https://docs.ferretdb.io/Basic_operations/read/#retrieve-all-documents-in-a-collection

**Review by w84thesun — APPROVED** [review]

File: website/docs/understanding_ferretdb.md (hunk 1 above)
Comment (AlekSi) [reply website/docs/understanding_ferretdb.md:85]: Ok, so what is the difference between that and the previous line? The one above requires `.md` to be absent, and the one below requires `.md` to be present?

Comment (AlekSi) [reply website/docs/understanding_ferretdb.md:85]: Ok, so what is the difference between that and the previous line? The one above requires `.md` to be absent, and the one below requires `.md` to be present?

**Review by AlekSi — COMMENTED** [review]
https://github.com/FerretDB/FerretDB/pull/1478#discussion_r1025118748

File: website/docs/understanding_ferretdb.md [hunk 2]
```diff
@@ -82,8 +82,8 @@ To reference the fourth field in the array, use the dot notation `"animals.3"`.
...
+* [Update an array](./basic_operations/update.md#update-an-array-element)
```
Comment (rumyantseva) [inline website/docs/understanding_ferretdb.md:86]: Hmm, here we have `.md` in the link text, but not in the other places? Why?

File: website/docs/understanding_ferretdb.md (hunk 1 above)
Comment (AlekSi) [reply website/docs/understanding_ferretdb.md:85]: @rumyantseva welcome to the club!

Comment (Fashander) [reply website/docs/understanding_ferretdb.md:85]: Yes, that's what I think. But if it strips out the extension when accessing the sections, then it wouldn't matter. Do let me know, thanks.

Absent: https://docs.ferretdb.io/Basic_operations/read/#retrieve-all-documents-in-a-collection
Present: https://docs.ferretdb.io/Basic_operations/read.md/#retrieve-all-documents-in-a-collection

File: website/docs/understanding_ferretdb.md [hunk 3]
```diff
@@ -115,9 +115,8 @@ To reference the `city` field in the embedded document, use the dot notation `"c
 
 For dot notation examples on embedded documents, see here:
 
-* Query an embedded document
-* Update an embedded document
-* Query an array of embedded documents
+* [Query an embedded document](./basic_operations/read.md#query-on-an-embedded-or-nested-document)
+* [Update an embedded document](./basic_operations/update.md#update-an-embedded-document)
```
Comment (rumyantseva) [inline website/docs/understanding_ferretdb.md:119]: And here we have `.md` in both places. What's the approach to it? When we do and don't add `.md` to the links?

File: website/docs/understanding_ferretdb.md (hunk 2 above)
Comment (Fashander) [reply website/docs/understanding_ferretdb.md:86]: Thanks, `.md` should be added.

**Review by rumyantseva — APPROVED** [review]
LGTM

**Review by AlekSi — DISMISSED** [review]

**Review by noisersup — APPROVED** [review]

**Review by w84thesun — APPROVED** [review]

**Review by rumyantseva — APPROVED** [review]
LGTM

**Comment by Fashander** [comment]
@AlekSi PTAL

**Review by chilagrow — APPROVED** [review]

**Review by AlekSi — APPROVED** [review]

**Pull request merged**